### PR TITLE
Add support for internal image store

### PR DESCRIPTION
**Pull request by FunkyM**
This PR implements a basic local image uploader to allow Grafana to act as an image store for images used in notifiers as discussed in #6922.

It was already used successfully in production for a while.

If merged, I'd also recommend to refactor/rename the "imgupoader" component into "imagestore" as I think it is a suitable name for arbitrary features around storing/handling generated images.

### CODE REVIEW

File: pkg/components/imguploader/imguploader.go
Comment (bergquist) [inline pkg/components/imguploader/imguploader.go]: Im not sure how this will work. Whats the purpose of the public path? How Should the user access the images over http? 

By default grafana saves rendered images to `data/png/` cant we just reuse that path?

**Review by bergquist — CHANGES_REQUESTED** [review]
Thank you for taking the time to contribute to Grafana!

Some questions before we merge this.

File: pkg/components/imguploader/imguploader.go
Comment (FunkyM) [reply pkg/components/imguploader/imguploader.go]: The idea was to make it configurable for arbitrary use-cases and just make it work for now.

Some reasoning:
- I didn't know how exactly the `data` directories are supposed to be used and I didn't want to actually expose them to the public frontend user as it seems logical that these are used by the backend.
- I was not sure if implementing the logic using `http_server` static routes or similar is the way to go.
- As `imageOnDiskPath` is passed, I can't assume `data/png` as the image path. Managing routes per passed images is also not an option.

So the simple solution was to just copy them to a custom folder that is accessible by the frontend.

My example configuration is this:
```
[external_image_storage.local]
path = ../data/storage
public_path = public/img/attachments
```

Do you have a suggestion how to prevent the file duplication here or if adding a static route to `data/png` is the way to go? I'd assume the optimal implementation would just expose the path with the generated images, not copy anything and not need any configuration.

File: pkg/components/imguploader/localuploader.go
Comment (FunkyM) [reply pkg/components/imguploader/localuploader.go]: You mean simply calling `in.Close()` after `out.Sync()`?

File: pkg/components/imguploader/localuploader.go
Comment (FunkyM) [reply pkg/components/imguploader/localuploader.go]: Should be `nil` as the closure should aready handle `out.Close()`, shouldn't it?

File: pkg/components/imguploader/localuploader_test.go
Comment (FunkyM) [reply pkg/components/imguploader/localuploader_test.go]: Sure, will enable them.

File: pkg/components/imguploader/imguploader.go
Comment (bergquist) [reply pkg/components/imguploader/imguploader.go]: Adding a route to the data/png should be enough. Using local image store should be a configurationless feature.

File: pkg/components/imguploader/localuploader.go
Comment (bergquist) [reply pkg/components/imguploader/localuploader.go]: Ignore that. Use defer to Close the file. Otherwise, we would not close it if `io.Copy` fails

**Comment by FunkyM** [comment]
Now uses a static route. Much better as it requires no configuration and just works.
Let me know if I should change it into a single commit without the rewrite and if anything else is wrong.

**Comment by FunkyM** [comment]
Squashed into a single commit since it makes sense.
Not sure how to fix the ci failed build though.

**Review by bergquist — APPROVED** [review]

**Comment by bergquist** [comment]
This looks much better! :) Thank you for taking the time.

**Comment by FunkyM** [comment]
@berquist Thanks, that fixed it. Please also let me know if you approve a refactoring of "imguploader" to an "imgstore" component. I could try to do that next.

**Comment by FunkyM** [comment]
@bergquist sorry to bug, any remaining things to do?

**Comment by bergquist** [comment]
@FunkyM thank you for your patience. This looks good. 

Thank you for contributing to Grafana! :)

**Comment by kerams** [comment]
It would be nice to also have the possibility to configure automatic cleanup of old images.

**Pull request merged**
